From 03898fd0c95aa1c7873bf09adc011d502f6a517d Mon Sep 17 00:00:00 2001
From: Dhananjay Kumar <dhakumar@codeaurora.org>
Date: Thu, 18 Apr 2019 21:52:36 +0530
Subject: [PATCH 21/38] audio: Add support for APTX TWSP audio codec

- Add support for APTX TWSP audio format for BT.

[wight554: updated code from LA.QSSI.11.0.r1-05600-qssi.0]

 Conflicts:
	media/java/android/media/AudioSystem.java
	services/core/java/com/android/server/audio/BtHelper.java

CRs-Fixed: 2433041
Change-Id: I19e256ae981516e9054b6e6eed8372ae2929b006
Signed-off-by: Volodymyr Zhdanov <wight554@gmail.com>
---
 media/java/android/media/AudioSystem.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/media/java/android/media/AudioSystem.java b/media/java/android/media/AudioSystem.java
index 85e12c08..0284eac0 100644
--- a/media/java/android/media/AudioSystem.java
+++ b/media/java/android/media/AudioSystem.java
@@ -202,6 +202,8 @@ public class AudioSystem
     public static final int AUDIO_FORMAT_CELT           = 0x26000000;
     /** @hide */
     public static final int AUDIO_FORMAT_APTX_ADAPTIVE  = 0x27000000;
+    /** @hide */
+    public static final int AUDIO_FORMAT_APTX_TWSP      = 0x2A000000;
 
     /** @hide */
     @IntDef(flag = false, prefix = "AUDIO_FORMAT_", value = {
@@ -231,6 +233,8 @@ public class AudioSystem
             case AUDIO_FORMAT_CELT: return BluetoothCodecConfig.SOURCE_CODEC_TYPE_CELT;
             case AUDIO_FORMAT_APTX_ADAPTIVE:
                      return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_ADAPTIVE;
+            case AUDIO_FORMAT_APTX_TWSP:
+                     return BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_TWSP;
             default:
                 Log.e(TAG, "Unknown audio format 0x" + Integer.toHexString(audioFormat)
                         + " for conversion to BT codec");
@@ -260,6 +264,8 @@ public class AudioSystem
                 return AudioSystem.AUDIO_FORMAT_CELT;
             case BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_ADAPTIVE:
                 return AudioSystem.AUDIO_FORMAT_APTX_ADAPTIVE;
+            case BluetoothCodecConfig.SOURCE_CODEC_TYPE_APTX_TWSP:
+                return AudioSystem.AUDIO_FORMAT_APTX_TWSP;
             default:
                 Log.e(TAG, "Unknown BT codec 0x" + Integer.toHexString(btCodec)
                         + " for conversion to audio format");
-- 
2.28.0

